src/customer/strategies/bolt/bolt-customer-strategy.ts

```diff
@@ -0,0 +1,119 @@
+
+    initialize(options: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options;
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const state = this._store.getState();
+        const paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+        if (!paymentMethod || !paymentMethod.initializationData.publishableKey) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const { developerConfig, publishableKey } = paymentMethod.initializationData;
+
+        return this._boltScriptLoader.load(publishableKey, paymentMethod.config.testMode, developerConfig)
+            .then(boltCheckout => this._boltClient = boltCheckout)
+            .then(() => this._store.getState());
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._boltClient = undefined;
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    continueAsGuest(credentials: GuestCredentials, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options || {};
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this.openCustomBoltCheckoutWidget(credentials.email, methodId)
+            .then(() => this._store.dispatch(this._billingAddressActionCreator.continueAsGuest(credentials, options)))
+            .then(() => this._store.getState());
+    }
+
+    signIn(credentials: CustomerCredentials, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options || {};
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._store.dispatch(this._customerActionCreator.signInCustomer(credentials, options))
+            .then(() => this.openCustomBoltCheckoutWidget(credentials.email, methodId))
+            .then(() => this._store.getState());
+    }
+
+    signOut(options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(this._customerActionCreator.signOutCustomer(options));
+    }
+
+    signUp(customerAccount: CustomerAccountRequestBody, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options || {};
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._store.dispatch(this._customerActionCreator.createCustomer(customerAccount, options))
+            .then(() => this.openCustomBoltCheckoutWidget(customerAccount.email, methodId))
+            .then(() => this._store.getState());
+    }
+
+    private openCustomBoltCheckoutWidget(email: string, methodId: string): Promise<InternalCheckoutSelectors> {
```

Comment:
We have a convention of prefixing private members, including methods, with an underscore. This is because when TS is compiled to JS, people know that they shouldn't be accessing those members.

Reply:
Fixed!